create-hpc-cluster/azuredeploy.json

```diff
@@ -76,121 +89,59 @@
-        "virtualNetworkName": "[concat(parameters('namePrefix'), 'VNet')]",
+        "storageAccountName": "[concat(parameters('clusterName'), 'sa')]",
+        "storageAccountId": "[resourceId('Microsoft.Storage/storageAccounts',variables('storageAccountName'))]",
+        "virtualNetworkName": "[concat(parameters('clusterName'), 'VNet')]",
         "addressPrefix": "10.0.0.0/16",
         "subnet1Name": "Subnet-1",
         "subnet1Prefix":"10.0.0.0/24",
         "vnetID": "[resourceId('Microsoft.Network/virtualNetworks',variables('virtualNetworkName'))]",
         "subnetRef": "[concat(variables('vnetID'),'/subnets/',variables('subnet1Name'))]",
         "publicIPAddressType":  "Dynamic",
-        "publicIPAddressName": "[concat(parameters('namePrefix'), '-PublicIP-VM')]",
+        "publicIPAddressName": "[concat(parameters('clusterName'), '-PublicIP-VM')]",
         "publicIPAddressNameHN": "[concat(variables('publicIPAddressName'),'-HN')]",
-        "dnsName": "[concat(parameters('namePrefix'), 'ep')]",
-        "domainNameHN": "[concat(variables('dnsName'),'-hn')]",
-        "nicName": "[concat(parameters('namePrefix'),'-nic')]",
+        "dnsName": "[concat(parameters('clusterName'), 'ep')]",
+        "nicName": "[concat(parameters('clusterName'),'-nic')]",
         "nicNameHN": "[concat(variables('nicName'),'-hn')]",
-        "vmNameHN": "[concat(parameters('namePrefix'),'hn')]",
-        "dcName": "[concat(parameters('namePrefix'),'dc.local')]",
         "vmExtensionName": "customScriptExtension",
-        "customScriptLocation": "http://yongjun.blob.core.windows.net/scripts/",
-        "iaasInfoArg": "[concat('-SubscriptionId ', subscription().subscriptionId, ' -VNet ', variables('virtualNetworkName'), ' -Subnet ', variables('subnet1Name'), ' -Location \"', parameters('location'), '\"')]",
-        "headNodeCommandStr": "[concat('powershell.exe -ExecutionPolicy Unrestricted -File PrepareHN.ps1 -DomainFQDN ', variables('dcName'), ' -AdminUserName ', parameters('adminUsername'), ' -AdminPassword \"', parameters('adminPassword'), '\" -PostConfigScript \"', parameters('headNodePostConfigScript'), '\" -NodePrepare ', variables('iaasInfoArg'))]",
-        "computeNodeCommandStr": "[concat('powershell.exe -ExecutionPolicy Unrestricted -File PrepareCN.ps1 -DomainFQDN ', variables('dcName'), ' -ClusterName ', variables('vmNameHN'),' -AdminUserName ', parameters('adminUsername'), ' -AdminPassword \"', parameters('adminPassword'), '\" -PostConfigScript \"', parameters('computeNodePostConfigScript'), '\"')]"
+        "availableCNImageSkus": {
+            "ComputeNode": "2012R2CN",
+            "ComputeNodeWithExcel": "2012R2CNExcel"
+        },
+        "computeNodeImageSku": "[variables('availableCNImageSkus')[parameters('computeNodeImage')]]",
+        "adminBase64Password": "[base64(parameters('adminPassword'))]",
+        "customScriptLocation": "https://suzhuhpcshare.blob.core.windows.net/scripts/",
```

Comment:
Currently we have problem to download the script from the github repo. I believe it is because we configured the DNS sever(i.e. the Domain Controller) for the VNET at the very beginning before it is available. We will fix this issue later. But before that, the scripts has to locate in the azure storage. After all, the scripts were put in my teammate's storage account, this checkin doesn't make worse.